thrifty-runtime/src/main/java/com/microsoft/thrifty/transport/URLTransport.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Thrifty
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING
+ * WITHOUT LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE,
+ * FITNESS FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ * See the Apache Version 2.0 License for specific language governing permissions and limitations under the License.
+ */
+package com.microsoft.thrifty.transport;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class URLTransport extends Transport {
+
+    private URL url;
+    private int timeout = 0;
+
+    private HttpURLConnection connection;
+    private InputStream inputStream;
+    private ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+    public static class Builder {
+
+        private URL url;
+        private int timeout = 0;
+
+        public Builder(URL url) {
+            this.url = url;
+        }
+
+        public Builder setTimeout(int timeout) {
```

Comment:
1. What unit is timeout - seconds, millis?  This should be document, and the param name should reflect it, e.g. `timeoutSeconds`.
2. What is this timeout - connect timeout, read timeout, etc?  This should also be documented, and the method name should reflect the purpose.